### ISSUE TITLE
Add option to start thelounge in background

### ISSUE THREAD

**Issue by yzgyyang** [thelounge/thelounge#2549]
* *Node version:* any
* *Browser version:* any
* *Device, operating system:* any
* *The Lounge version:* any

---

I understand that for platforms that have prebuilt packages, by using `systemctl` `service(8)` or similar, thelounge will be up and running as a service in the background. However, for users who install thelounge via npm/yarn, extra steps are needed to have it run in background.

IMO it would be really useful to have a CLI option to configure thelounge to start then move to the background, something like:
```
thelounge start --background
```

It would be even better if we have the whole "service"-like option built in, like:
```
thelounge start
thelounge stop
thelounge restart
```

I haven't dived in the codebase too much, and it is just my own thought, I am sorry if this is not a feasible / needed feature.

**Comment by xPaw** [comment]
Yeah we need some kind of forking mechanism because that also aids us with other things.

If we store pidfile of lounge in a file, we can interact with it from other commands:

- Be able to signal the process that something changed (installed a new theme or package, changed config, etc)

**Comment by astorije** [comment]
I understand the need to be able to install packages or reload themes on the fly, but I'm not entirely sure if we want to implement and maintain a full-featured service manager.

Like you said, there is systemd and the likes that we provide built-in support with via the prebuilt packages, and when installing it from npm or source, there is a plethora of options out there such as good old `screen`, or Node-based solutions like `pm2` which take little-to-no setup (for `pm2`, a simple `pm2 start ...` is the only command you have to type, no configuration required).

I'm obviously not opposed to considering it, but I want to make sure we're not spreading ourselves even thinner and get something too complicated if it isn't actually necessary.

Welcome to The Lounge, @yzgyyang! :)

**Comment by McInkay** [comment]
So, this isn't a replacement for things like systemd, it's just that instead of running in the foreground, we fork into the background, and we have a way to communicate across multiple instances of the lounge. It doesn't replace or do the same thing as a service manager, nor does the service manager do the things requested. I think it's a fair suggestion, and would be a really nice addition.

I wouldn't say we make it a priority, but if someone wants to tackle it we definitely let them. There are libraries that help work with this as well.

**Comment by tracker1** [comment]
I think a better route might be to have a service installer as part of the npm package... unfortunately, I haven't seen any great service installer options that support Linux, Windows and macOS.  Most tend to require a compiled binary build step even.  It might be worth exploring over having a launch in background.

**Comment by xPaw** [comment]
We have a debian package which creates a service.

**Comment by Azrael25** [comment]
Hi People, Sorry if this isnt the right place to ask this, topic is related.. how do you get thelounge to stay open in the background? when i start it via Thelounge & then  thelounge start -c port=9000 -c public=true
it will start but when i exit via Ctrl C it shuts down thelounge? can someone point me in the right direction?

thanks.. and once again sorry if this is the wrong place to ask!

**Comment by yzgyyang** [comment]
@Azrael25 You could either use tmux or write a service yourself :)
